dapps/shop/backend/utils/validators.js

```diff
@@ -0,0 +1,41 @@
+function assert(cond, message) {
+  if (!cond) {
+    throw new Error(`Assertion error: ${message}`)
+  }
+  return true
+}
+
+/**
+ * Validate a Shop object for creating a new shop
+ */
+function validateShop(shop) {
+  try {
+    assert(typeof shop === 'object', 'Invalid shop object')
+    assert(typeof shop.name === 'string', 'Shop name should be a string')
+    assert(shop.name !== '', 'Shop name missing')
+    assert(shop.listing_id, 'Shop needs a listing ID')
```

Comment:
Should this be set in the config? Config seems to hold network specific options and this is one

Reply:
Every shop is dapp deployment specific which is network specific.

Reply:
You mean we don't care about the network id? Could we just put the `stripe_backend` and `data_url` settings as columns here instead of in the config object in that case?

Reply:
Network ID is cared about.  Each multi-tenant backend is network specific. The deployed one is Rinkeby, for instance.

Don't really follow about the other config.  Listing ID was specifically added here as a column because the listener needed to do lookups based on that ID.  I'd rather not move more from config to columns without a good reason.

Reply:
Was looking at the DB again today and noticed there's also other network ID references that are kind of superfluous. Not really sure what they're from(e.g. the Networks table, and `network_id` on a handful of tables).  Don't think the listener was ever made to work with multiple networks and the DB files are named using the network name(e.g. `net_rinkeby.db`).

Should probably clean this up at some point.